src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private static final int MAX_CAPACITY = 1 << 30;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (table == null) {
+            table = new Node[INITIAL_CAPACITY];
+        }
+        int i = index(hash(key));
+        Node<K, V> currentNode = table[i];
+        int threshold = (int) (table.length * LOAD_FACTOR);
+        if (currentNode == null) {
+            table[i] = newNode;
+            if (++size > threshold) {
+                resize();
+            }
+            return;
+        }
+        do {
+            if (keyCheck(key, currentNode)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                if (++size > threshold) {
+                    resize();
+                }
+                return;
+            }
+            currentNode = currentNode.next;
+        } while (true);
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
```

Comment:
We wont need this check if we initialize table in constructor

Reply:
Got rid of it.